common/src/main/java/com/wynntils/models/seaskipper/SeaskipperModel.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.seaskipper;
+
+import com.google.common.reflect.TypeToken;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ContainerSetSlotEvent;
+import com.wynntils.mc.event.MenuEvent;
+import com.wynntils.mc.event.ScreenOpenedEvent;
+import com.wynntils.models.containers.ContainerModel;
+import com.wynntils.models.items.items.gui.SeaskipperDestinationItem;
+import com.wynntils.models.map.pois.SeaskipperPoi;
+import com.wynntils.screens.maps.SeaskipperMapScreen;
+import com.wynntils.utils.mc.ComponentUtils;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public final class SeaskipperModel extends Model {
+    private static final StyledText OAK_BOAT_NAME = StyledText.fromString("§bOak Boat");
+    private static final Pattern SEASKIPPER_PASS_PATTERN = Pattern.compile("^§b(.*) Pass §7for §b(\\d+)²$");
+    private int containerId = -2;
+    private static SeaskipperMapScreen currentScreen;
+
+    public SeaskipperModel(ContainerModel containerModel) {
+        super(List.of(containerModel));
```

Comment:
Call loadSeaskipperPois() here instead, and keep all possible destinations as a part of the model.

Reply:
I can't call it here as when the Seaskipper poi's get created they need the player location to determine whether the player is inside and this gets called before I even join the server

Reply:
You need to split this up and rewrite the logic a bit. See my comments further down the file.